File: backend/app/schemas/review_items.py
from pydantic import BaseModel, Field


class ReviewItemCreateRequest(BaseModel):
    finding_id: int
    review_reason: str = Field(min_length=1)
    assigned_owner_user_id: str = Field(min_length=1)
    due_date: str = Field(min_length=1)
    created_by_user_id: str | None = None


class ReviewItemCreateResponse(BaseModel):
    review_item_id: int
    finding_id: int
    review_status: str
    assigned_owner_user_id: str
    due_date: str
    created_utc: int


class ReviewItemUpdateRequest(BaseModel):
    review_status: str = Field(min_length=1)
    resolution_note: str | None = None
    defer_reason: str | None = None


class ReviewItemUpdateResponse(BaseModel):
    review_item_id: int
    review_status: str
    updated_utc: int
    resolution_note: str | None
    defer_reason: str | None


class ReviewItemDetailResponse(BaseModel):
    review_item_id: int
    finding_id: int
    analysis_id: str
    application_name: str
    finding_headline: str
    kb_reference: str
    review_reason: str
    assigned_owner_user_id: str
    due_date: str
    review_status: str
    created_utc: int
    updated_utc: int
    created_by_user_id: str
    resolution_note: str | None
    defer_reason: str | None

File: frontend/src/services/reviewItems.ts
import { apiPatch, apiPost } from "./api";

export type ReviewItemCreateRequest = {
  finding_id: number;
  review_reason: string;
  assigned_owner_user_id: string;
  due_date: string;
  created_by_user_id?: string;
};

export type ReviewItemCreateResponse = {
  review_item_id: number;
  finding_id: number;
  review_status: string;
  assigned_owner_user_id: string;
  due_date: string;
  created_utc: number;
};

export type ReviewItemUpdateRequest = {
  review_status: string;
  resolution_note?: string;
  defer_reason?: string;
};

export type ReviewItemUpdateResponse = {
  review_item_id: number;
  review_status: string;
  updated_utc: number;
  resolution_note: string | null;
  defer_reason: string | null;
};

export function createReviewItem(payload: ReviewItemCreateRequest) {
  return apiPost<ReviewItemCreateResponse, ReviewItemCreateRequest>(
    "/review-items",
    payload,
  );
}

export function updateReviewItem(reviewItemId: number, payload: ReviewItemUpdateRequest) {
  return apiPatch<ReviewItemUpdateResponse, ReviewItemUpdateRequest>(
    `/review-items/${reviewItemId}`,
    payload,
  );
}